spec/controllers/webhook_headers_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "rails_helper"
+
+RSpec.describe WebhookHeadersController, type: :controller do
+  let(:valid_session) { {} }
+  let(:registry) { create(:registry) }
+  let(:user) { create(:user) }
+  let(:viewer) { create(:user) }
+  let(:contributor) { create(:user) }
+  let(:owner) { create(:user) }
+  let(:team) do
+    create(:team,
+           owners:       [owner],
+           viewers:      [user, viewer],
+           contributors: [contributor])
+  end
+  let(:namespace) do
+    create(
+      :namespace,
+      team:        team,
+      description: "short test description",
+      registry:    registry)
+  end
+  let(:webhook) { create(:webhook, namespace: namespace) }
+
+  before :each do
+    # trigger creation of registry
+    registry
```

Comment:
Just don't :smile:. Simply create the registry with `let!` instead of `let` previously. The same should go for the rest of the variables defined in `let` blocks. For what I've seen in other places in our own tests, using `let` ends up with "funny surprises". For this, I prefer using `let!` as much as possible, even if it's slower than `let`.